src/components/MultiIndexResults.js

```diff
@@ -0,0 +1,71 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import cx from 'classnames';
+
+import Template from './Template';
+
+class MultiIndexResults extends React.Component {
+  static propTypes = {
+    cssClasses: PropTypes.shape({
+      root: PropTypes.string,
+      item: PropTypes.string,
+      empty: PropTypes.string,
+    }),
+    derivedIndices: PropTypes.array,
+    templateProps: PropTypes.object.isRequired,
+  };
+
+  renderNoResults({ label }) {
+    const { cssClasses: { root, empty } } = this.props;
+    const className = cx(root, empty);
+    return (
+      <div key={label}>
+        <strong>{label}</strong>
+        <Template
+          rootProps={{ className }}
+          templateKey="empty"
+          {...this.props.templateProps}
+        />
+      </div>
+    );
+  }
+
+  renderResults({ label, hits }) {
+    const { cssClasses: { item } } = this.props;
+
+    const hitsMarkup = hits.map((hit, position) => {
+      const data = { ...hit, __hitIndex: position };
```

Comment:
probably the same in regular hits, but if you put it already as `__hitIndex` in the `map`, there's no need to rename